ui/app/store/actions.js

```diff
@@ -389,11 +389,18 @@ export function removeAccount (address) {
     dispatch(showLoadingIndication())
 
     return new Promise((resolve, reject) => {
-      background.removeAccount(address, (err, account) => {
+      background.removeAccount(address, async (error, account) => {
+        try {
+          await forceUpdateMetamaskState(dispatch)
+        } catch (error) {
+          dispatch(displayWarning(error.message))
+          return reject(error)
```

Comment:
If we return here, do we not hide the loading indicator?

Reply:
Ah.... excellent point.

Reply:
I didn't like how this turned out anyway. I've rewritten it to use `try ... catch ... finally` and I think it handles this case correctly now, plus it reads better.